src/ray/common/ray_config_def.h

```diff
@@ -78,7 +78,19 @@ RAY_CONFIG(uint64_t, raylet_check_gc_period_milliseconds, 100)
 
 /// Threshold when the node is beyond the memory capacity.
 /// Ranging from [0, 1]
-RAY_CONFIG(float, memory_usage_threshold_fraction, 0.9)
+///
+/// The memory threshold is calculcated as
+/// Memory usage threshold = max(node_memory * memory_usage_threshold_fraction,
+/// node_memory - max_overhead_bytes)
+RAY_CONFIG(float, memory_usage_threshold_fraction, 0.95)
+
+/// The maximum amount of memory overhead to keep before we consider the node running low
+/// on memory. Disabled if it is -1.
+///
+/// The memory threshold is calculcated as
+/// Threshold = max(node_memory * memory_usage_threshold_fraction, node_memory -
+/// max_overhead_bytes)
+RAY_CONFIG(int64_t, max_overhead_bytes, (int64_t)2 * 1024 * 1024 * 1024)
```

Comment:
Maybe min_memory_free is a better name?

Reply:
I think it's implied that it will be best-effort. For someone with no context, min_memory_free or min_memory_free_bytes seems the most easily understandable. max_memory_free doesn't seem accurate since we will often have much more free memory than this value. Not sure that "overhead" is clear enough on its own.

Reply:
Addressed comments and move to 

https://github.com/ray-project/ray/pull/29111/files